lib/model/model.go

```diff
@@ -2432,42 +2441,33 @@ func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly
 
 		f.Name = strings.Replace(f.Name, prefix, "", 1)
 
-		var dir, base string
-		if f.IsDirectory() && !f.IsSymlink() {
-			dir = f.Name
-		} else {
-			dir = filepath.Dir(f.Name)
-			base = filepath.Base(f.Name)
-		}
-
 		if levels > -1 && strings.Count(f.Name, sep) > levels {
 			return true
 		}
 
-		last := output
-		if dir != "." {
-			for _, path := range strings.Split(dir, sep) {
-				directory, ok := last[path]
-				if !ok {
-					newdir := make(map[string]interface{})
-					last[path] = newdir
-					last = newdir
-				} else {
-					last = directory.(map[string]interface{})
-				}
-			}
+		fullName := strings.Split(f.Name, sep)
+		name := fullName[len(fullName)-1]
+		path := fullName[:len(fullName)-1]
+
+		tree := DirectoryTree{Name: name, IsDirectory: f.IsDirectory(), Children: []*DirectoryTree{}}
```

Comment:
no point allocating the children slice if it's not a directory.

Reply:
That is true. However, this:

```
var children []*FolderItem = nil
if f.FileType() == 1 {
	children = make([]*FolderItem, 0)
}
tree := FolderItem{Name: name, FileType: f.FileType(), Children: children}
``` 
is much longer and less readable that this:
tree := FolderItem{Name: name, FileType: f.FileType(), Children:[]*FolderItem{}}

Isn't the capacity in the second case allocated to 0 anyway? Just asking. I will use the `if` if you think it would be better.

Reply:
Array with the size of zero is not the same as no array, as there is still array struct allocated with size and capacity.

In theory, you can just leave it as nil in all cases because when you do:

`foo.Children = append(foo.Children, newChild)`

if foo.Children is nil, the append would allocate a new array for you.

Reply:
Ah so. Now I understand. I'll leave `nill` there.